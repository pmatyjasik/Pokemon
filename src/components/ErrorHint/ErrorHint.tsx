import { Typography, useTheme } from "@mui/material";

export interface ErrorHintProps {
    error?: string;
}

export const ErrorHint = ({ error }: ErrorHintProps) => {
    const { palette } = useTheme();

    return error ? (
        <Typography variant='caption' align='left' color={palette.error.main}>
            {error}
        </Typography>
    ) : null;
};
