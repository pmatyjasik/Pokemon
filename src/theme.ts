"use client";

import { createTheme, Shadows } from "@mui/material/styles";
import { ThemeOptions } from "@mui/material/styles";
import { NextFont } from "next/dist/compiled/@next/font";
import localFont from "next/font/local";

const IbmVga: NextFont = localFont({
    src: "./fonts/IBM_VGA.woff",
    display: "swap",
    style: "normal",
    weight: "400",
});

const themeOptions: ThemeOptions = {
    palette: {
        mode: "light",
        background: {
            default: "rgba(255, 255, 255, 1)",
        },
        primary: {
            main: "rgba(151, 71, 255, 1)",
            dark: "rgba(113, 53, 191, 1)",
            light: "rgba(151, 71, 255, 0.25)",
        },
        grey: {
            100: "rgba(42, 42, 42, 1)",
            200: "rgba(127, 127, 127, 1)",
            300: "rgba(228, 228, 228, 1)",
            400: "rgba(238, 238, 238, 1)",
        },
        error: {
            main: "rgba(255, 78, 78, 1)",
        },
        action: {
            disabled: "rgba(0, 0, 0, 0.2)",
        },
    },
    spacing: 4,
    typography: {
        fontFamily: IbmVga.style.fontFamily,
        fontWeightMedium: 400,
        fontSize: 16,
        h6: {
            fontSize: "12px",
        },
        caption: {
            fontSize: "10px",
        },
    },
    shape: {
        borderRadius: 2,
    },
    shadows: [
        "none",
        "0px 0px 0px 4px rgba(151, 71, 255, 0.25)",
        "0px 4px 10px 2px rgba(0, 0, 0, 0.1)",
        ...Array(20).fill("none"),
    ] as Shadows,
};

export const theme = createTheme(themeOptions);
