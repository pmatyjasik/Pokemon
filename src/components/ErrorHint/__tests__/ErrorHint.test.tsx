import { cleanup, render, screen } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import { ErrorHint, ErrorHintProps } from "@/components/ErrorHint/ErrorHint";
import { theme } from "@/theme";

const renderComponent = (props: ErrorHintProps) => {
    return render(
        <ThemeProvider theme={theme}>
            <ErrorHint {...props} />
        </ThemeProvider>
    );
};

const error = "Error message";

const defaultProps = {
    error,
} satisfies ErrorHintProps;

describe("ErrorHint component", () => {
    afterEach(() => {
        cleanup();
    });

    it("Matches DOM Snapshot", () => {
        const { asFragment } = renderComponent(defaultProps);

        expect(asFragment()).toMatchSnapshot();
    });

    test("renders nothing when error is undefined", () => {
        renderComponent({});

        const errorText = screen.queryByText(/./);

        expect(errorText).toBeNull();
    });

    test("renders error text when error is provided", () => {
        renderComponent(defaultProps);

        const errorText = screen.getByText(error);

        expect(errorText).toBeInTheDocument();
        expect(errorText).toHaveStyle(`color: ${theme.palette.error.main}`);
    });

    test("renders error text with 'caption' typography", () => {
        renderComponent(defaultProps);

        const errorText = screen.getByText(error);

        expect(errorText).toBeInTheDocument();
        expect(errorText.tagName.toLowerCase()).toBe("span");
    });
});
