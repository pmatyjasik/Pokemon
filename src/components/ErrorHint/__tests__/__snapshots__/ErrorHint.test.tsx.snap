// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorHint component Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiTypography-root MuiTypography-caption MuiTypography-alignLeft css-13dzput-MuiTypography-root"
  >
    Error message
  </span>
</DocumentFragment>
`;
